spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithoutReordering.java

```diff
@@ -238,5 +244,6 @@ public Type primitive(Type.PrimitiveType primitive) {
           .put(TypeID.STRING, ImmutableSet.of(StringType$.class))
           .put(TypeID.FIXED, ImmutableSet.of(BinaryType$.class))
           .put(TypeID.BINARY, ImmutableSet.of(BinaryType$.class))
+          .put(TypeID.VARIANT, ImmutableSet.of(VariantType$.class))
```

Comment:
The only place where this map (`TYPES`) is referenced is in the `primitive` method that isn't used for variants, so I don't think that this change is needed.